app/models/broken_window/metric.rb

```diff
@@ -8,17 +8,17 @@ class Metric < ActiveRecord::Base
 
     has_many :measurements
 
-    validates :name, presence: true, length: {maximum: 255}
-    validates :calculator, format: {
-      with: VALID_CLASS_NAME, allow_blank: true, message: "must be a valid class name" }
-    validates :value_type, inclusion: {in: VALUE_TYPES}
-    validates :threshold, presence: true, numericality: {allow_blank: true}, unless: :container?
-    validates :threshold_type, inclusion: {in: THRESHOLD_TYPES}
+    validates :name, presence: true, length: { maximum: 255 }
+    validates :calculator, format: { with: VALID_CLASS_NAME, allow_blank: true, message: "must be a valid class name" }
+    validates :value_type, inclusion: { in: VALUE_TYPES }
+    validates :threshold, presence: true, numericality: { allow_blank: true }, unless: :container?
+    validates :threshold_period, presence: true, format: { with: /\A[+-]?\d+\z/, message: "should be integer" }, numericality: { allow_blank: true, greater_than_or_equal_to: 1 }, unless: :container?
```

Comment:
could you please remove "format" from here? numericality already does that, the more code you add the longer it is to read.